src/incubator/Slider/index.tsx

```diff
@@ -0,0 +1,575 @@
+      } else if (locationX === rangeThumbOffset.value.x) {
+        activeThumb.value = ThumbType.DEFAULT;
+        updateDefaultThumb(locationX);
+      } else if (locationX > rangeThumbOffset.value.x) {
+        activeThumb.value = ThumbType.RANGE;
+        updateRangeThumb(locationX);
+      } else if (locationX < defaultThumbOffset.value.x) {
+        activeThumb.value = ThumbType.DEFAULT;
+        updateDefaultThumb(locationX);
+      } else if (locationX > defaultThumbOffset.value.x && locationX < rangeThumbOffset.value.x) {
+        if (activeThumb.value === ThumbType.DEFAULT) {
+          updateDefaultThumb(locationX);
+        } else {
+          updateRangeThumb(locationX);
+        }
+      }
+    } else {
+      updateDefaultThumb(locationX);
+    }
+  }, []);
+
+  /** gestures and animations */
+
+  const activeThumb = useSharedValue(ThumbType.DEFAULT);
+
+  const isPressedDefault = useSharedValue(false);
+  const defaultThumbOffset = useSharedValue({x: 0, y: 0});
+  const defaultThumbStart = useSharedValue({x: 0, y: 0});
+
+  const isPressedRange = useSharedValue(false);
+  const rangeThumbOffset = useSharedValue({x: 0, y: 0});
+  const rangeThumbStart = useSharedValue({x: 0, y: 0});
+
+  const defaultThumbAnimatedStyles = useAnimatedStyle(() => {
+    const activeStyle = 
+      isPressedDefault.value ? _activeThumbStyle.value : _thumbStyle.value;
+    return {
+      transform: [
+        {translateX: (defaultThumbOffset.value.x - thumbSize.value.width / 2) * rtlFix},
+        {scale: withSpring(!disableActiveStyling && isPressedDefault.value ? 1.3 : 1)}
+      ],
+      top: -(thumbSize.value.height - trackSize.value.height) / 2,
+      ...activeStyle
+    };
+  });
+
+  const rangeThumbAnimatedStyles = useAnimatedStyle(() => {
+    const activeStyle = 
+      isPressedDefault.value ? _activeThumbStyle.value : _thumbStyle.value;
+    return {
+      transform: [
+        {translateX: (rangeThumbOffset.value.x - thumbSize.value.width / 2) * rtlFix},
+        {scale: withSpring(!disableActiveStyling && isPressedRange.value ? 1.3 : 1)}
+      ],
+      top: -(thumbSize.value.height - trackSize.value.height) / 2,
+      ...activeStyle
+    };
+  });
+
+  const updateDefaultThumb = (x: number) => {
+    defaultThumbOffset.value = {
+      x,
+      y: 0
+    };
+    defaultThumbStart.value = {
+      x,
+      y: 0
+    };
+
+    activeTrackWidth.value = Math.abs(useRange ? rangeThumbStart.value.x - x : x);
+    
+    onChange(useRange ? {min: x, max: rangeThumbStart.value.x} : x);
+  };
+
+  const updateRangeThumb = (x : number) => {
+    rangeThumbOffset.value = {
+      x,
+      y: 0
+    };
+    rangeThumbStart.value = {
+      x,
+      y: 0
+    };
+    
+    activeTrackWidth.value = x - defaultThumbStart.value.x;
+    
+    onChange({min: defaultThumbStart.value.x, max: x});
+  };
+
+  const defaultThumbGesture = Gesture.Pan()
+    .onBegin(() => {
+      if (disabled) {
+        return;
+      }
```

Comment:
Consider instead calling the `enabled` method on the gesture like this 
```
defaultThumbGesture.enabled(!disabled);
```

Reply:
Cool. Thanks